jupyterhub_singleuser_profiles/profiles.py

```diff
@@ -296,11 +298,31 @@ def apply_gpu_config(self, gpu_mode, gpu_count, pod):
     return pod
 
   @classmethod
-  def apply_pod_profile(self, spawner, pod, profile):
+  def generate_volume_path(self, mountPath, default_mount_path, volume_name):
+    if (mountPath):
+      if (os.path.isabs(mountPath)):
+        return mountPath
+      return os.path.join(default_mount_path, mountPath)
+    return os.path.join(default_mount_path, volume_name)
+    
+
+  @classmethod
+  def apply_pod_profile(self, spawner, pod, profile, default_mount_path):
     api_client = kubernetes.client.ApiClient()
 
     pod.metadata.labels['jupyterhub.opendatahub.io/user'] = escape(spawner.user.name)
 
+    profile_volumes = profile.get('volumes')
+
+    if profile_volumes:
+      for volume in profile_volumes:
+        volume_name = re.sub('[^a-zA-Z0-9\.]', '_', volume['name'])
```

Comment:
Seems like we need to use `-` instead of `_` - 

```
Invalid value: \"data_et\": a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name', or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?'```